src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -380,6 +382,8 @@ private void Cancel()
                 {
                     _waitSource.SetResult(true);
                 }
+
+                if (HttpTelemetry.IsEnabled) HttpTelemetry.Log.RequestStop();
```

Comment:
Not Abort?

Reply:
I agree about Abort. It also looks like this is outside of a lock. Does this risk double-counting RequestStop?

Reply:
I've been chasing a few bugs right now with this PR about correct numbers of 'Start' and 'Stop' events. I'll verify this and potentially add some synchronization for `RequestStop` to avoid double-counting.